Lesson21/Kubra/app.js

```diff
@@ -3,49 +3,63 @@
 // load - all elements (including images and styles) are loaded
 window.addEventListener('load',function(){
   // Select all images
-  const images= document.querySelectorAll("img"); // Array of image
+  const images = document.querySelectorAll("img"); // Array of image
   // Select previous and next buttons
   const previousButton = document.getElementById("prev-btn");
-  const nextButton =document.getElementById("next-btn");
+  const nextButton = document.getElementById("next-btn");
   // Select indicators
-  const indicators = document.getElementsByClassName('indicator');
-
-
+  const indicators = document.querySelectorAll('.indicator');
   // Create application state to keep track of the current index
   let currentIndex = 0;
-
+  
   // Show images function
-  function showImage(index) {
+  function showImage() {
     // Remove active class from all images
-    images.forEach(
-      // For every element in images array call a function that removes "active" class
-    )
-
+    images.forEach((img) => {
+      img.classList.remove('active');
+    }); 
+    // For every element in images array call a function that removes "active" class
     // Remove active class from all indicators
+    indicators.forEach((indicator) => {
+      indicator.classList.remove('active');
+    });
     // Use loop to loop through all element of array
 
     // Add active class to the current image and indicator
     // images[]
-  }
+      images[currentIndex].classList.add("active");  
+      indicators[currentIndex].classList.add('active');
+  };
 
   // handle next click
   function showNextImage() {
     // increase currentIndex. What happens if the current index is > than number of images that we have?
     currentIndex++;
+    if(currentIndex >= images.length) {
+      currentIndex = 0;
+    };
     showImage(currentIndex);
   }
 
   // handle previous click
   function showPrevImage() {
+    currentIndex--;
+    if(currentIndex < 0){
+      currentIndex = images.length - 1;
+    };
+    showImage(currentIndex);
     // [image1, image2, image3]
     // decrease currentIndex. What happens if the current index is < 0?
     // if else condition
   }
 
   // Add event listeneters for button clicks
-
+  nextButton.addEventListener('click', showNextImage);
+  previousButton.addEventListener('click', showPrevImage);
+  
 
   // Switch images after 3 seconds automatically using setInterval function
   // Use 3000 value for 3 second delay
-
+  
+  let timer = setInterval(showNextImage, 3000);
```

Comment:
You don't need to assign `setInterval(showNextImage, 3000);` to a variable, you only need to call it once.
```suggestion
  setInterval(showNextImage, 3000);
```

Reply:
Actually, I know that it isn't necessary. The reason I assign to a var is that I was planning to add 'mouseover' and 'mouseout' to make it better. Kind of annoying when mouseover the picture it still continues auto skip. But it was late and I was tired at the end. I also think that maybe it won't be a good idea to add something extra which you didn't ask. Anyway:)) I am going to fix now. Thank you for your feedback!

Reply:
I think it's actually a really good idea, feel free to add it and keep this as your own project, I'm pretty sure you can reuse this code in the future. :)